pyiceberg/table/inspect.py

```diff
@@ -657,3 +717,19 @@ def all_manifests(self) -> "pa.Table":
             lambda args: self._generate_manifests_table(*args), [(snapshot, True) for snapshot in snapshots]
         )
         return pa.concat_tables(manifests_by_snapshots)
+
+    def position_deletes(self) -> "pa.Table":
```

Comment:
nit: for any metadata tables referencing the current snapshot, let's add an optional param to allow querying for any given snapshot id.